students/JeroenSweerts/Lesson3/string_formatting_exercise.py

```diff
@@ -0,0 +1,62 @@
+#!/usr/bin/env python3
+
+###################TASK 1#######################################################
+'''Write a format string that will take the following four element tuple:
+( 2, 123.4567, 10000, 12345.67)
+and produce:
+"file_002 :   123.46, 1.00e+04, 1.23e+04"  '''
+tuple_elements = ( 2, 123.4567, 10000, 12345.67)
+#print('file_{0:03d} :   {}'.format(tuple_elements[0],tuple_elements[1]))
+print('file_{0:03d} :  {1:.2f}, {2:.2e}, {3:.2e}'.format(tuple_elements[0],tuple_elements[1],tuple_elements[2],tuple_elements[3]))
+
+###################TASK 2#######################################################
+'''Using your results from Task One, repeat the exercise,
+but this time using an alternate type of format string
+(hint: think about alternative ways to use .format() (keywords anyone?),
+and also consider f-strings if you’ve not used them already).'''
+print(f'file_{tuple_elements[0]:03d} :  {tuple_elements[1]:.2f}, {tuple_elements[2]:.2e}, {tuple_elements[3]:.2e}')
+
+###################TASK 3#######################################################
+def formatter(*t):
+    form_string = ""
+    for el in t[0]:
+        form_string = form_string + "{:d},"
+    #delete last comma of form_string
+    form_string = form_string[:-1]
+
+    print("the "+str(len(*t))+" numbers are: " +form_string.format(*t[0]))
```

Comment:
I would generally try to stick with one type of string concat, it makes it a bit harder to read with different flavors.

either ony f-string or format would work well here.